examples/travel/add_things_to_do_ad.rb

```diff
@@ -0,0 +1,257 @@
+    customer_id: customer_id,
+    operations: [ad_group_operation]
+  )
+
+  # Fetch the new ad group's resource name.
+  ad_group_resource = response.results.first.resource_name
+
+  puts "Added an ad group with resource name '#{ad_group_resource}'."
+
+  ad_group_resource
+end
+# [END add_things_to_do_ad_2]
+
+# Creates a new ad group ad in the specified ad group
+# [START add_things_to_do_ad_3]
+def add_ad_group_ad(client, customer_id, ad_group_resource)
+  # Creates a new ad group ad and sets a travel ad info.
+  ad_group_ad_operation = client.operation.create_resource.ad_group_ad do |aga|
+    aga.ad = client.resource.ad do |ad|
+      ad.travel_ad = client.resource.travel_ad_info
+    end
+    # Set the ad group ad to enabled. Setting this to paused will cause an error
+    # for Things to do campaigns. Pausing should happen at either the ad group
+    # or campaign level.
+    aga.status = :ENABLED
+
+    # Set the ad group.
+    aga.ad_group = ad_group_resource
+  end
+
+  # Issue a mutate request to add the ad group ad.
+  ad_group_ad_service = client.service.ad_group_ad
+  response = ad_group_ad_service.mutate_ad_group_ads(
+    customer_id: customer_id,
+    operations: [ad_group_ad_operation],
+  )
+
+  # Fetch the new ad group ad's resource name.
+  ad_group_ad_resource = response.results.first.resource_name
+
+  puts "Added an ad group ad with resource name '#{ad_group_ad_resource}'."
+end
+# [END add_things_to_do_ad_3]
+
+
+# Appends a random number to the provided description text and returns it as a
+# string-wrapped value
+def generate_random_name_field(text)
+  "#{text} ##{(Time.new.to_f * 100).to_i}"
+end
+
+
+if __FILE__ == $0
+  options = {}
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_id] = 'INSERT_CUSTOMER_ID_HERE'
+  options[:things_to_do_center_account_id] = 'INSERT_THINGS_TO_DO_CENTER_ACCOUNT_ID_HERE'
+
+  OptionParser.new do |opts|
+    opts.banner = sprintf('Usage: %s [options]', File.basename(__FILE__))
+    opts.separator ''
+    opts.separator 'Options:'
+    opts.on('-C', '--customer-id CUSTOMER-ID', String, 'Customer ID') do |v|
+      options[:customer_id] = v
+    end
+
+    opts.on('-T', '--things-to-do-center-account-id THINGS-TO-DO-CENTER-ACCOUNT-ID',
+      Integer, 'Things To Do Account ID') do |v|
+      options[:things_to_do_center_account_id] = v
+    end
+
+    opts.separator ''
+    opts.separator 'Help:'
+
+    opts.on_tail('-h', '--help', 'Show this message') do
+      puts opts
+      exit
+    end
+  end.parse!
+
+  begin
+    add_things_to_do_ad(options.fetch(:customer_id).tr("-", ""), options[:things_to_do_center_account_id])
```

Comment:
Use "fetch" for mandatory fields (which I think things_to_do_center_account_id is), since it'll result in an error if no key is set.

Reply:
Yes, its required. Changed it to fetch.